src/kvstore/raftex/RaftPart.cpp

```diff
@@ -382,22 +371,34 @@ folly::Future<AppendLogResult> RaftPart::appendLogAsync(ClusterID source,
                 retFuture = cachingPromise_.getSharedFuture();
                 break;
         }
-        if (replicatingLogs_) {
-            VLOG(2) << idStr_
-                    << "Another AppendLogs request is ongoing,"
-                       " just return";
-            return retFuture;
-        } else {
+
+        bool expected = false;
+        if (replicatingLogs_.compare_exchange_strong(expected, true)) {
```

Comment:
replicatingLogs_ is protected by logsLock_, maybe no need CAS?

Reply:
replicatingLogs_ has no relationship with logsLock_.   We can't assume it will be accessed under protection by logsLock_,  actually we accessed it in many places directly without logsLock_